core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -2291,6 +2286,11 @@ private int addColumnToMeta(
                 nameOffset += Vm.getStorageLength(columnName);
             }
             ddlMem.putStr(name);
+
+            final int commitMode = configuration.getCommitMode();
+            if (commitMode != CommitMode.NOSYNC) {
+                ddlMem.sync(commitMode == CommitMode.ASYNC);
```

Comment:
We should always sync metadata files if we want better stability on power loss.

Reply:
same comment as above, with NOSYNC we lose column data quite easily

Reply:
on a second thought, this is column add, may as well sync always: https://github.com/questdb/questdb/pull/3304/commits/f9de70ee205e8d2de8b792cfb10fc8db4b9c1110